Main.lean

```diff
@@ -81,8 +143,8 @@ def main : IO Unit := do
   IO.println "* * *"
 
   IO.println "* * *"
-  IO.println "(i32.add (i32.const 42)) is represented as:"
-  void $ parseTestP addP "(i32.add (i32.const 42))"
+  IO.println "i32.add (i32.const 42) is represented as:"
```

Comment:
Currently, this parses as "add from_operation from_stack". I think it's fine like that, but we can think about it.

Reply:
Well, no, not really. This code has a different meaning.

It's only fine if you KNOW you're running it from opsP. 

I think, this is a bug. I mean, it sucks that every op parser will have the stupid Chars.between, but life is life. You can turn `Chars.between "(" ")"` into a combinator!

Reply:
I disagree pretty heavily. By construction, _we_ will only run this from `opsP`, and if your concern is for the Wasm.lean users, nothing is stopping them from blowing things up by defining something similar even if the built-in stuff is parenthesised.
Additionally, this is IMO good scaffolding if we ever do decide to support full instructions, and actually, the way we parse it, it matters very little whether we parse it in parentheses or not, because either way there are two – no more, no less – "holes" that are filled with either `from_operation` or `from_stack`.

I think adding the parentheses combinator to each operation parser is just unnecessary visual clutter that adds nothing in terms of safety.

Reply:
If we decide to support pure instructions, we'll need to write another parser!

I agree with all your arguments, but then we'll have to make all these parsers private.

Frankly, I don't care much, but your approach *will* be more confusing for the readers of `addP` who have no context.

Anyway, we can write something like `NB! This parses add OPERATIONS only, not INSTRUCTIONS.  Operations are S-expression thingies, whereas instructions are raw asm thingies.`

Smt like this.